services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeServiceImpl.kt

```diff
@@ -150,6 +131,40 @@ open class HankeServiceImpl(
         }
     }
 
+    private fun createEntityFromCreateRequest(
```

Comment:
Just an idea. Could we have this as a non private function somewhere? It would simplify testing (hanke creation based on request) and provide a single source of truth on how an entity is created from a request. Currently this part is not needed anywhere else, but if it was, this would be valuable.

Or is there a reason that this has to be private?

Reply:
This method doesn't build the complete entity, since things like hanketunnus, generated, perustaja, yhteystiedot and tormaystarkastelu are handled in the parent function. Because of the limited functionality, I think it would be misleading to make this public. The visibility can be adjusted later, of course, if the need arises. But I don't expect it will.

Also, because of the call to hanketunnusService, this can be only run in integration tests without extra mocking. In integration tests, creating an entity that's not backed in the DB would be weird. So it's hard to see the value this method would have in tests.

Reply:
Fair enough.